app2/src/lib/services/transfer-ucs03-evm/state.ts

```diff
@@ -0,0 +1,72 @@
+import { Data, type Exit, type Effect } from "effect"
+import type { Hash } from "viem"
+import type { submitTransfer, waitForTransferReceipt } from "./transactions.ts"
+import type { switchChain } from "./chain.ts"
+import type { waitForApprovalReceipt } from "$lib/services/transfer-ucs03-evm/approval.ts"
+
+type EffectToExit<T> = T extends Effect.Effect<infer A, infer E, any> ? Exit.Exit<A, E> : never
+
+export type SwitchChainState = Data.TaggedEnum<{
+  InProgress: {}
+  Complete: { exit: EffectToExit<ReturnType<typeof switchChain>> }
+}>
+export const SwitchChainState = Data.taggedEnum<SwitchChainState>()
+
+export type ApprovalSubmitState = Data.TaggedEnum<{
+  InProgress: {}
+  Complete: {
+    exit: EffectToExit<ReturnType<typeof switchChain>> // TODO: change function
```

Comment:
Now that we have a function for this, you can change it to `typeof approveTransfer`